lib/stud/task.rb

```diff
@@ -36,5 +36,14 @@ def stop!
     def self.interrupted?
       Thread.current[:stud_task_interrupted]
```

Comment:
I think I have introduced a bug here, It should be @thread[] and not `Thread.current`, I have to check the Stud::Interval code more closely.